nova/bigvm/manager.py

```diff
@@ -307,10 +307,19 @@ def _get_providers(self, context):
                               {'rp': rp['uuid']})
                     continue
                 inventory = resp.json()["inventories"]
-                memory_mb_inventory = inventory[MEMORY_MB]
+                # Note(jakobk): It's possible to encounter incomplete (e.g.
+                # in-buildup) resource providers here, that don't have all the
+                # usual resources set.
+                memory_mb_inventory = inventory.get(MEMORY_MB)
+                if not memory_mb_inventory:
+                    LOG.info('no MEMORY_MB_RESOURCE in RP %(rp)s',
```

Comment:
`no MEMORY_MB resource` looks like it would fit more as we never have a `MEMORY_MB_RESOURCE` and the variable also isn't named like thtat.